code/game/objects/structures/signs/_signs.dm

```diff
@@ -4,137 +4,177 @@
+		if(!do_after(user, 4 SECONDS, target = src))
+			return
+		var/obj/structure/sign/sign_type = GLOB.editable_sign_types[choice]
+		name = initial(sign_type.name)
+		desc = "[initial(sign_type.desc)] It can be placed on a wall."
+		icon_state = initial(sign_type.icon_state)
+		sign_path = sign_type	
+		user.visible_message("<span class='notice'>You finish changing the sign.</span>")
+		return
+	return ..()
 
 /obj/item/sign_backing
 	name = "sign backing"
-	desc = "A sign with adhesive backing. Use a pen to change the decal once installed."
+	desc = "A plastic sign with adhesive backing, use a pen to change the decal. It can be placed on a wall."
 	icon = 'icons/obj/decals.dmi'
 	icon_state = "backing"
 	w_class = WEIGHT_CLASS_NORMAL
 	custom_materials = list(/datum/material/plastic = 2000)
+	armor = list("melee" = 50, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 50, "acid" = 50)
 	resistance_flags = FLAMMABLE
-	var/sign_path = /obj/structure/sign/basic //the type of sign that will be created when placed on a turf
+	max_integrity = 100
+	var/sign_path = /obj/structure/sign/basic //The type of sign structure that will be created when placed on a turf, the default looks just like a sign backing item.
+	var/is_editable = TRUE
+
+/obj/item/sign_backing/Initialize() //Signs not attached to walls are always rotated so they look like they're laying horizontal.
+	. = ..()
+	var/matrix/M = matrix()
+	M.Turn(90)
+	transform = M
 
 /obj/item/sign_backing/afterattack(atom/target, mob/user, proximity)
 	. = ..()
-	if(isturf(target) && proximity)
-		var/turf/T = target
-		user.visible_message("<span class='notice'>[user] fastens [src] to [T].</span>", \
-							 "<span class='notice'>You attach the sign to [T].</span>")
-		playsound(T, 'sound/items/deconstruct.ogg', 50, TRUE)
-		var/obj/structure/sign/S = new sign_path(T)
-		S.setDir(dir)
-		qdel(src)
+	if(!iswallturf(target) || !proximity)
+		return
+	var/turf/T = target
+	var/turf/userT = get_turf(user)
+	var/obj/structure/sign/S = new sign_path(userT) //We place the sign on the turf the user is standing, and pixel shift it to the target wall, as below.
+	//This is to mimic how signs and other wall objects are usually placed by mappers, and so they're only visible from one side of a wall.
+	var/dir = get_dir(userT, T)
+	switch(dir)
+		if(NORTH)
+			S.pixel_y = 32
+		if(SOUTH)
+			S.pixel_y = -32
+		if(EAST)
+			S.pixel_x = 32
+		if(WEST)
+			S.pixel_x = -32
+		else //Signs cannot be placed diagonally.
+			to_chat(user, "<span class='warning'>Your reach is unsteady, stand directly in front of the wall you wish to place a sign on")
+			qdel(S)
+			return 
+	user.visible_message("<span class='notice'>[user] fastens [src] to [T].</span>", \
+						 "<span class='notice'>You attach the sign to [T].</span>")
+	playsound(T, 'sound/items/deconstruct.ogg', 50, TRUE)
+	S.obj_integrity = obj_integrity
+	S.setDir(dir)
+	qdel(src)
```

Comment:
If somebody knows a better way to do this, PLEASE tell me. Issue is placing signs diagonally and easily figuring out what their pixel shift should be.